src/main/java/core/basesyntax/figures/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax.figures;
+
+import java.util.Locale;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double topSide;
+    private double bottomSide;
+    private double lateralSide;
+    private double area;
+
+    public IsoscelesTrapezoid(String color, double topSide, double bottomSide, double lateralSide) {
+        super(color);
+        this.topSide = topSide;
+        this.bottomSide = bottomSide;
+        this.lateralSide = lateralSide;
+        this.area = calculateArea();
+    }
+
+    public double getTopSide() {
+        return topSide;
+    }
+
+    public void setTopSide(double topSide) {
+        this.topSide = topSide;
+        this.area = calculateArea();
```

Comment:
Consider whether recalculating the area upon setting a new side is the best approach. It could lead to performance issues if these setters are called frequently. It might be better to calculate the area on-demand in the `getArea()` method.